java/client/src/main/java/glide/api/models/commands/bitmap/BitFieldOptions.java

```diff
@@ -0,0 +1,197 @@
+
+        /** Specifies if the offset uses encoding multiplier. */
+        private final BitOffset offset;
+
+        /** Value to increment the bits in the binary value. */
+        private final long increment;
+
+        public String[] toArgs() {
+            return new String[] {
+                "INCRBY", encoding.getEncoding(), offset.getOffset(), Long.toString(increment)
+            };
+        }
+    }
+
+    /**
+     * <code>OVERFLOW</code> subcommand that determines the result of the <code>SET</code> or <code>
+     * INCRBY</code> commands when an under or overflow occurs.
+     */
+    @RequiredArgsConstructor
+    public static final class BitFieldOverflow implements BitFieldSubCommands {
+        /** Overflow behaviour. */
+        private final BitOverflowControl overflowControl;
+
+        public String[] toArgs() {
+            return new String[] {"OVERFLOW", overflowControl.toString()};
+        }
+
+        /** Supported bit overflow controls */
+        public enum BitOverflowControl {
+            /**
+             * Performs modulo when overflow occurs with unsigned encoding. When overflows occur with
+             * signed encoding, the value restart towards the most negative value. When underflows occur
+             * with signed, the value restart towards the most positive ones.
+             */
+            WRAP,
+            /** Underflows set to the minimum value and overflows sets to the maximum value. */
+            SAT,
+            /** Returns null when overflows occur. */
+            FAIL
+        }
+    }
+
+    /** Specifies if the argument is a signed or unsigned encoding */
+    private interface BitEncoding {
+        String getEncoding();
+    }
+
+    /** Specifies that the argument is a signed encoding */
+    public static final class SignedEncoding implements BitEncoding {
+        @Getter private final String encoding;
+
+        /**
+         * Constructor that prepends the number with "i" to specify that it is in signed encoding.
+         *
+         * @param encodingLength bit size of encoding.
+         */
+        public SignedEncoding(long encodingLength) {
+            encoding = "i".concat(Long.toString(encodingLength));
+        }
+    }
+
+    /** Specifies that the argument is a signed encoding */
+    public static final class UnsignedEncoding implements BitEncoding {
+        @Getter private final String encoding;
+
+        /**
+         * Constructor that prepends the number with "u" to specify that it is in unsigned encoding.
+         *
+         * @param encodingLength bit size of encoding.
+         */
+        public UnsignedEncoding(long encodingLength) {
+            encoding = "u".concat(Long.toString(encodingLength));
+        }
+    }
+
+    /** Offset in the array of bits. */
+    private interface BitOffset {
+        String getOffset();
+    }
+
+    /** Offset in the array of bits. */
+    public static final class Offset implements BitOffset {
```

Comment:
I just recognized that I don't understand from which size offset is measured. Can you add these details to docs?
If I have `01010010` and getting `u4` with offset `0`, `1` and `#1` - what I get?
Consider adding examples here - but as a lower priority task.